ui/components/multichain/create-named-snap-account/create-named-snap-account.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import { InternalAccount } from '@metamask/keyring-api';
+import { setAccountLabel } from '../../../store/actions';
+import { CreateAccount } from '..';
+import { Box, ModalHeader } from '../../component-library';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { getMostRecentOverviewPage } from '../../../ducks/history/history';
+import { getAccountName, getKeyringSnapAccounts } from '../../../selectors';
+
+export type CreateNamedSnapAccountProps = {
+  /**
+   * Callback called once the account has been created
+   */
+  onActionComplete: (completed: boolean, reject?: boolean) => Promise<void>;
+
+  /**
+   * Address of the account to create
+   */
+  address: string;
+
+  /**
+   * Suggested account name from the snap
+   */
+  snapSuggestedAccountName?: string;
+};
+
+export const CreateNamedSnapAccount: React.FC<CreateNamedSnapAccountProps> = ({
+  onActionComplete,
+  address,
+  snapSuggestedAccountName,
+}) => {
+  const t = useI18nContext();
+  const history = useHistory();
+  const dispatch = useDispatch();
+
+  const mostRecentOverviewPage = useSelector(getMostRecentOverviewPage);
+
+  const onCreateAccount = useCallback(async (name: string) => {
+    dispatch(setAccountLabel(address, name));
+    await onActionComplete(true);
+  }, []);
+
+  const getNextAccountName = useCallback(
+    async (accounts: InternalAccount[]): Promise<string> => {
+      // if snapSuggestedAccountName exists, return it immediately
+      if (snapSuggestedAccountName) {
+        return snapSuggestedAccountName;
+      }
+
+      const defaultAccountName: string = getAccountName(accounts, address);
+      // if defaultAccountName is truthy, return it immediately
+      if (defaultAccountName) {
+        return defaultAccountName;
+      }
+
+      const snapAccounts = getKeyringSnapAccounts();
+      const accountNumber = snapAccounts.length;
+      return `Snap Account ${accountNumber}`;
```

Comment:
Don't we miss a `+ 1` here? If we have 1 Snap account like `['Snap Account 1']`, `.length` will be `1` so the returned value here would be `Snap Account 1` again, while it should be `Snap Account 2`, no?

Reply:
The account has been temporarily added to the keyring so the potential new account will be the last one. 
Flow is:
Temporarily add the new account to keyring -> show naming modal (temp account present as last added snap account) -> persist the account or remove it if canceled

Reply:
Then, we should add a comment explaining why we are not taking the next one here, cause this feels a bit non-natural to me 😅 WDYT?

Reply:
good idea! [87b706f](https://github.com/MetaMask/metamask-extension/pull/25191/commits/87b706f1496daafadc9a8279673136d805fe25d1)